packages/repository/src/repositories/kv.repository.bridge.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as legacy from 'loopback-datasource-juggler';
+
+import {Options, DataObject} from '../common-types';
+import {Model} from '../model';
+
+import {KVRepository, AsyncIterator, KVFilter} from './kv.repository';
+
+import {juggler, ensurePromise} from './legacy-juggler-bridge';
+
+/**
+ * An implementation of KVRepository based on legacy loopback-datasource-juggler
+ */
+export class DefaultKVRepository<T extends Model> implements KVRepository<T> {
+  kvModelClass: typeof juggler.KeyValueModel;
+
+  constructor(kvModelClass: typeof juggler.KeyValueModel) {
+    this.kvModelClass = kvModelClass;
+  }
+
+  delete(key: string, options?: Options): Promise<void> {
+    return ensurePromise(this.kvModelClass.delete(key, options));
+  }
+
+  deleteAll(options?: Options): Promise<void> {
+    return ensurePromise(this.kvModelClass.deleteAll(options));
+  }
+
+  get(key: string, options?: Options): Promise<T> {
+    const val = this.kvModelClass.get(key, options) as legacy.PromiseOrVoid<T>;
```

Comment:
This is not a valid cast. Typically, `T` is a subclass of the new LB4 `Model` class, while `kvModelClass.get` returns an instance of legacy LB3 `KeyValueModel` class. (See also my comment above.)

Reply:
> This is not a valid cast. Typically, `T` is a subclass of the new LB4 `Model` class, while `kvModelClass.get` returns an instance of legacy LB3 `KeyValueModel` class. (See also my comment above.)

☝️ this is an important point to address or discuss ☝️ 

You can take a look at how (CRUD) Repository deals with conversion between backing-model instances and LB4 model instances.